### PR TITLE
Add TestSuite handling to FancyTestdoxPrinter

### LINKED ISSUE #19: FancyTestdoxPrinter does not handle errors in setUpBeforeClass
Executing the following test with the FancyTestdoxPrinter
```
<?php

namespace rpkamp;

use Exception;
use PHPUnit\Framework\AssertionFailedError;
use PHPUnit\Framework\TestCase;
use PHPUnit\Framework\Warning;

class FailureTest extends TestCase
{
    public static function setUpBeforeClass()
    {
        $a = new \Does\Not\Exist();
        // throw new Exception('What we have here is a failure to communicate');
    }

    /**
     * @test
     */
    public function does_nothing() {
        $this->assertEquals(true, true);
    }
}
```
Results in the following
```
PHPUnit 6.2.0 by Sebastian Bergmann and contributors.


Fatal error: Uncaught Error: Call to a member function fail() on null in /usr/src/app/src/FancyTestdoxPrinter.php on line 124

Error: Call to a member function fail() on null in /usr/src/app/src/FancyTestdoxPrinter.php on line 124

Call Stack:
    0.0135     355968   1. {main}() /usr/src/app/vendor/phpunit/phpunit/phpunit:0
    0.1370     687688   2. PHPUnit\TextUI\Command::main(???) /usr/src/app/vendor/phpunit/phpunit/phpunit:53
    0.1370     687800   3. PHPUnit\TextUI\Command->run(???, ???) /usr/src/app/vendor/phpunit/phpunit/src/TextUI/Command.php:141
    0.2152    2329056   4. PHPUnit\TextUI\TestRunner->doRun(???, ???, ???) /usr/src/app/vendor/phpunit/phpunit/src/TextUI/Command.php:210
    0.2184    2502112   5. PHPUnit\Framework\TestSuite->run(???) /usr/src/app/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:541
    0.2216    2508440   6. PHPUnit\Framework\TestSuite->run(???) /usr/src/app/vendor/phpunit/phpunit/src/Framework/TestSuite.php:746
```

The cause of this issue is that `TestSuite` is passed to [startTest](https://github.com/rpkamp/fancy-testdox-printer/blob/master/src/FancyTestdoxPrinter.php#L59) and [return](https://github.com/rpkamp/fancy-testdox-printer/blob/master/src/FancyTestdoxPrinter.php#L62) is called. This means the [currentTestResult](https://github.com/rpkamp/fancy-testdox-printer/blob/master/src/FancyTestdoxPrinter.php#L94-L98) is not created. The subsequent call to [addError](https://github.com/rpkamp/fancy-testdox-printer/blob/master/src/FancyTestdoxPrinter.php#L122-L128) fails because `$this->currentTestResult` is `null`.

### PR DESCRIPTION
**Pull request by samlitowitz**
Resolves #19

### CODE REVIEW

File: src/FancyTestdoxPrinter.php [hunk 1]
```diff
@@ -89,6 +92,9 @@ public function startTest(Test $test)
         } elseif ($test instanceof PhptTestCase) {
             $className  = $class;
             $testMethod = $test->getName();
+        } elseif ($test instanceof TestSuite) {
+            $className = $class;
+            $testMethod = "setUpBeforeClass";
```
Comment (rpkamp) [inline src/FancyTestdoxPrinter.php:97]: Can this never be any other method? Seems tricky to hardcode this.

**Comment by rpkamp** [comment]
I just solved this same problem for PHPUnit 7, where this printer is part of, and I have to say your solution is _much_ more elegant than mine. Might have to revisit that :smiley:

File: src/FancyTestdoxPrinter.php (hunk 1 above)
Comment (samlitowitz) [reply src/FancyTestdoxPrinter.php:97]:
~~~This problem only occurs when there are errors in static functions I suspect. I will double check this.~~~ This is exclusively for `setUpBeforeClass`.

**Comment by rpkamp** [comment]
Awesome, thanks @samlitowitz !

**Pull request merged**